packages/flutter/lib/src/widgets/image.dart

```diff
@@ -930,11 +917,22 @@ class Image extends StatefulWidget {
   /// If non-null, this color is blended with each image pixel using [colorBlendMode].
   final Color? color;
 
-  /// Used to set the [FilterQuality] of the image.
+  /// The rendering quality of the image.
+  ///
+  /// If the image is of a high quality and its pixels are perfectly aligned
+  /// with the physical screen pixels, extra quality enhancement may not be
+  /// necessary. If so, then [FilterQuality.none] would be the most efficient.
```

Comment:
Check with Skia, but I believe that they might already detect this case and optimize it.

I'm not sure if the same could be said of the html web-renderer, though.